packages/@coorpacademy-components/src/atom/button-link/index.tsx

```diff
@@ -2,16 +2,46 @@ import React, {useCallback} from 'react';
 import {noop} from 'lodash/fp';
 import classnames from 'classnames';
 import Link from '../link';
+import FaIcon from '../icon';
 import {ICONS} from '../../util/button-icons';
 import propTypes, {ButtonLinkProps, IconType} from './types';
 import style from './style.css';
 
-const getButtonContent = (
-  icon?: IconType & {customStyle?: Record<string, unknown>},
-  label?: string
-) => {
-  const {type, position, customStyle} = icon || {type: '', position: ''};
+const getButtonContent = (icon?: IconType, label?: string) => {
+  const {type, faIconName, faIconColor, faIconBackgroundColor, faIconSize, position, customStyle} =
+    icon || {type: '', position: ''};
   const Icon = type && ICONS[type];
+  const FaIconParsed: any = FaIcon;
+
+  if (faIconName) {
```

Comment:
seems like it's the same component here,
I think you can define the component one time before the return, like so:
```
const faIconComponent = (
      <FaIconParsed
        iconName={faIconName}
        iconColor={faIconColor}
        backgroundColor={faIconBackgroundColor}
        size={{
          faSize: faIconSize,
          wrapperSize: faIconSize
        }}
      />
    );
```
and use it after for both case below